packages/cli/commands/project/migrateApp.js

```diff
@@ -0,0 +1,210 @@
+const path = require('path');
+const {
+  addAccountOptions,
+  addConfigOptions,
+  getAccountId,
+  addUseEnvironmentOptions,
+} = require('../../lib/commonOpts');
+const { trackCommandUsage } = require('../../lib/usageTracking');
+const { loadAndValidateOptions } = require('../../lib/validation');
+const {
+  createProjectPrompt,
+} = require('../../lib/prompts/createProjectPrompt');
+const { i18n } = require('../../lib/lang');
+const {
+  fetchPublicAppOptions,
+  selectPublicAppPrompt,
+} = require('../../lib/prompts/selectPublicAppPrompt');
+const { poll } = require('../../lib/polling');
+const {
+  uiBetaTag,
+  uiLine,
+  uiLink,
+  uiCommandReference,
+  uiAccountDescription,
+} = require('../../lib/ui');
+const SpinniesManager = require('../../lib/ui/SpinniesManager');
+const {
+  logApiErrorInstance,
+  ApiErrorContext,
+} = require('../../lib/errorHandlers/apiErrors');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { promptUser } = require('../../lib/prompts/promptUtils');
+const { isAppDeveloperAccount } = require('../../lib/accountTypes');
+const { ensureProjectExists } = require('../../lib/projects');
+const {
+  migrateApp,
+  checkMigrationStatus,
+} = require('@hubspot/local-dev-lib/api/projects');
+const { getCwd } = require('@hubspot/local-dev-lib/path');
+const { logger } = require('@hubspot/local-dev-lib/logger');
+const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
+const { downloadProject } = require('@hubspot/local-dev-lib/api/projects');
+const { extractZipArchive } = require('@hubspot/local-dev-lib/archive');
+const { getHubSpotWebsiteOrigin } = require('@hubspot/local-dev-lib/urls');
+
+const i18nKey = 'commands.project.subcommands.migrateApp';
+
+exports.command = 'migrate-app';
+exports.describe = null; // uiBetaTag(i18n(`${i18nKey}.describe`), false);
+
+exports.handler = async options => {
+  await loadAndValidateOptions(options);
+
+  const accountId = getAccountId(options);
+  const accountConfig = getAccountConfig(accountId);
+  const accountName = uiAccountDescription(accountId);
+
+  trackCommandUsage('migrate-app', {}, accountId);
+
+  if (!isAppDeveloperAccount(accountConfig)) {
+    logger.error(
+      i18n(`${i18nKey}.errors.invalidAccountType`, {
+        accountName,
+        accountType: accountConfig.accountType,
+        useCommand: uiCommandReference('hs accounts use'),
+        authCommand: uiCommandReference('hs auth'),
+      })
+    );
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  const { appId } =
+    'appId' in options
+      ? options
+      : await selectPublicAppPrompt({
+          accountId,
+          accountName,
+          options,
+          migrateApp: true,
+        });
+
+  const validateAppId = async (appId, accountId, accountName) => {
```

Comment:
Looks like you could probably just remove this function and run the next few lines inline if its not used anywhere else

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1052/commits/17da9c8463e4edcfc9166abc3ad864a30736e0dd